examples/resource_group_declarative_api.py

```diff
@@ -0,0 +1,157 @@
+        requests={"node_num": 0}, # this resource group can hold 0 nodes, no data will be load on it.
+        limits={"node_num": 10000}, # this resource group can hold at most 10000 nodes 
+    ))
+
+    # create a default resource group, which can used to hold the nodes that all initial node in it.
+    utility.update_resource_groups({
+        DEFAULT_RESOURCE_GROUP: ResourceGroupConfig(
+            requests={"node_num": node_num},
+            limits={"node_num": node_num},
+            transfer_from=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover missing node from pending resource group at high priority.
+            transfer_to=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover redundant node to pending resource group at low priority.
+        )})
+
+def list_all_resource_groups():
+    rg_names = utility.list_resource_groups()
+
+    for rg_name in rg_names:
+        resource_group = utility.describe_resource_group(rg_name)
+        print(resource_group)
+        # print(f"Resource group {rg_name} has {resource_group.nodes} with config: {resource_group.config}")
+
+def scale_resource_group_to(name :str, node_num: int):
+    """scale resource group to node_num nodes, new query node need to be added from outside orchestration system"""
+    utility.update_resource_groups({
+        name: ResourceGroupConfig(
+            requests={"node_num": node_num},
+            limits={"node_num": node_num},
+            transfer_from=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover missing node from pending resource group at high priority.
+            transfer_to=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover redundant node to pending resource group at low priority.
+        )
+    })
+
+def create_resource_group(name: str, node_num: int):
+    print(f"Create resource group {name} with {node_num} nodes")
+    utility.create_resource_group(name, config=ResourceGroupConfig(
+        requests={"node_num": node_num},
+        limits={"node_num": node_num},
+        transfer_from=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover missing node from pending resource group at high priority.
+        transfer_to=[{"resource_group": _PENDING_NODES_RESOURCE_GROUP}], # recover redundant node to pending resource group at low priority.
+    ))
+
+def resource_group_management():
+    # cluster is initialized with 1 node in default resource group, and 0 node in pending resource group.
+    init_cluster(1)
+    list_all_resource_groups()
+    # DEFAULT_RESOURCE_GROUP: 1
+    # _PENDING_NODES_RESOURCE_GROUP: 0
+
+    # rg1 missing two query node.
+    create_resource_group("rg1", 2)
+    list_all_resource_groups()
+    # DEFAULT_RESOURCE_GROUP: 1
+    # _PENDING_NODES_RESOURCE_GROUP: 0
+    # rg1: 0(missing 2)
+
+    # scale_out(2)
+    # scale out two new query node into cluster by orchestration system, these node will be added to rg1 automatically.
```

Comment:
These two comments are a little confusing.

Reply:
`scale_out` operator cannot be done by milvus itself.
Orchestration system like `k8s` is required.